aten/src/ATen/core/jit_type.h

```diff
@@ -259,9 +259,19 @@ struct CAFFE2_API OptionalType: public SingleElementType<TypeKind::OptionalType,
     return create(contained_types[0]);
   }
 
+  bool isSubtypeOf(const TypePtr rhs) const override {
+    if (Type::isSubtypeOf(rhs)) {
+      return true;
+    }
+    if(auto rhs_ = rhs->cast<OptionalType>()) {
+      return getElementType()->isSubtypeOf(rhs_->getElementType());
+    }
+    return false;
+  }
+
   bool operator==(const Type& rhs) const override {
     if (auto opt_rhs = rhs.cast<OptionalType>()) {
-      return getElementType()->isSubtypeOf(opt_rhs->getElementType());
+      return getElementType() == opt_rhs->getElementType();
```

Comment:
ah I was looking at this and thinking the operator looked wrong. I tried changing it to this locally but it didn't fix my issue -- didn't realize we'd need the new isSubtypeOf as well to get it to work.

Reply:
Yup I think the `isSubtypeOf` will fix your problem